Quadriga/src/main/java/edu/asu/spring/quadriga/service/workbench/mapper/impl/ProjectShallowMapper.java

```diff
@@ -240,4 +240,50 @@ public IProject getProjectDetails(ProjectDTO projectDTO) throws QuadrigaStorageE
 		return projectList;
 	}
 	
+	/**
+	 * {@inheritDoc}
+	*/
+	@Override
+	@Transactional
+	public IProject getProjectDetailsForSearch(ProjectDTO projectDTO, String pattern) throws QuadrigaStorageException{
+		IProject projectProxy = null;
+		if(projectDTO != null){
+			projectProxy = new ProjectProxy(projectManager);
+			projectProxy.setProjectId(projectDTO.getProjectid());
+			projectProxy.setProjectName(projectDTO.getProjectname());
+			projectProxy.setDescription(searchLines(projectDTO.getDescription(), pattern));

```

Comment:
this will result in deleting this method. You shouldn't have to change the mapper classes at all. Here you have 99% code duplication, which is unnecessary.